File: OLD_SRC_2/util/network/IServerPacket.java
package com.sleepwalker.sleeplib.util.network;

import net.minecraftforge.fml.network.NetworkDirection;

import javax.annotation.Nonnull;

public interface IServerPacket extends INetworkPacket {

    @Nonnull
    @Override
    default NetworkDirection getNetworkDirection(){
        return NetworkDirection.PLAY_TO_CLIENT;
    }
}

File: src/main/kotlin/com/sleepwalker/sleeplib/util/network/IServerPacket.kt
package com.sleepwalker.sleeplib.util.network

import net.minecraftforge.api.distmarker.Dist
import net.minecraftforge.api.distmarker.OnlyIn
import net.minecraftforge.fml.network.NetworkDirection
import net.minecraftforge.fml.network.NetworkEvent

interface IServerPacket : INetworkPacket {
    override val networkDirection: NetworkDirection
        get() = NetworkDirection.PLAY_TO_CLIENT

    override fun handle(context: NetworkEvent.Context) {
        handleClient(context)
    }

    @OnlyIn(Dist.CLIENT)
    fun handleClient(context: NetworkEvent.Context)
}